### PR TITLE
APPDEV-1113 Take device-specific links from the device entity

### PR DESCRIPTION
**Pull request by vishal-revdiwala**
- Created new model class for Device entity and accordingly code update
   Currently following links consumed for iOS: `appleMobileConfig` and `postOpenAppEvent`
- Code refactor

### CODE REVIEW

File: Yona/Yona/UserRequestManager.swift [hunk 1]
```diff
@@ -75,7 +75,14 @@ class UserRequestManager{
     
     
     func postOpenAppEvent(_ user: Users, onCompletion: @escaping APIResponse) {
-        if let path = user.openAppEventLink, let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"]  as? String, let appVersionCode = Bundle.main.infoDictionary?["CFBundleVersion"]  as? String {
+        var openAppEventLinkPath: String?
```
Comment (Bert-R) [inline Yona/Yona/UserRequestManager.swift]: This, along with the below for-loop should be reducible to something like this:
```
let openAppEventLinkPath = user.devices.first(where: { $0.isRequestingDevice }).map {$0.openAppEventLink}
```

File: Yona/Yona/UserRequestManager.swift [hunk 2]
```diff
@@ -210,9 +217,15 @@ class UserRequestManager{
     }
     
     func getMobileConfigFile( _ onCompletion: @escaping APIMobileConfigResponse) {
-       if let mobileConfigURL = self.newUser?.mobilConfigFileURL {
-           APIServiceManager.sharedInstance.callRequestWithAPIMobileConfigResponse(nil, path: mobileConfigURL, httpMethod: httpMethods.get, onCompletion: onCompletion)
+        var mobileConfigFileURL: String?
```
Comment (Bert-R) [inline Yona/Yona/UserRequestManager.swift]: Similar here.

File: Yona/Yona/UserRequestManager.swift (hunk 1 above)
Comment (vishal-revdiwala) [reply Yona/Yona/UserRequestManager.swift]: Done.

File: Yona/Yona/UserRequestManager.swift (hunk 2 above)
Comment (vishal-revdiwala) [reply Yona/Yona/UserRequestManager.swift]: Updated here too.

File: Yona/Yona/UserRequestManager.swift [hunk 3]
```diff
@@ -75,16 +75,10 @@ class UserRequestManager{
...
-            if temp.isRequestingDevice! {
-                openAppEventLinkPath = temp.openAppEventLink
-            }
-        }
-        
+        let openAppEventLinkPath = user.devices.first(where: { $0.isRequestingDevice! }).map {$0.openAppEventLink}
         if let path = openAppEventLinkPath, let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"]  as? String, let appVersionCode = Bundle.main.infoDictionary?["CFBundleVersion"]  as? String {
             let bodyForOpenAppEvent = ["operatingSystem": "IOS", "appVersion": appVersion, "appVersionCode":appVersionCode] as BodyDataDictionary
-            genericUserRequest(httpMethods.post, path: path, userRequestType: userRequestTypes.postUser, body: bodyForOpenAppEvent, onCompletion: { (success, message, code, nil) in
+            genericUserRequest(httpMethods.post, path: path!, userRequestType: userRequestTypes.postUser, body: bodyForOpenAppEvent, onCompletion: { (success, message, code, nil) in
```
Comment (Bert-R) [inline Yona/Yona/UserRequestManager.swift]: Just curious: 
You changed this
```
path: path
```

into this:
```
path: path!
```

What is the functional difference of that exclamation mark?

Comment (vishal-revdiwala) [reply Yona/Yona/UserRequestManager.swift]: Exclamation mark used in Swift to unwrapped the optional values. 
Now the type of `path` is a string, not an optional anymore.

**Pull request merged**